### PR TITLE
Task 3.2: pdfminer fallback extraction

### PR DESCRIPTION
**Pull request by Frank-Yong**
## Summary
- add `extract_pdfminer_chunk` fallback extraction in `docdown/stages/extract.py`
- add `PdfMinerError` with clear failure messages for missing chunk files, empty output, and extraction exceptions
- write fallback output as UTF-8 text to `.txt` artifacts and log extraction timing
- add unit tests for pdfminer success, empty output failure, and error wrapping
- update Task 3.2 checklist and add artifact class diagram

## Testing
- `pytest tests/test_extract.py -q`
- `pytest -q`

## Notes
- pdfminer fallback is implemented in Stage 2 extraction module for use by extraction orchestration in the next task.

### CODE REVIEW

File: docdown/stages/extract.py [hunk 1]
```diff
@@ -163,6 +168,47 @@ def extract_grobid_chunk(
...
+
+    chunk_path = Path(chunk_pdf)
+    output_path = Path(output_text)
+
+    active_logger: logging.Logger | ChunkAdapter
+    if chunk_number is not None:
+        active_logger = ChunkAdapter(logger, {"chunk": chunk_number}) if logger else get_chunk_logger(chunk_number)
+    else:
+        active_logger = logger or get_logger()
+
```
[the remark this answers is not in the document]
Comment (Frank-Yong) [reply docdown/stages/extract.py]: I refactored the shared logger selection; next I’m running the extraction tests to verify no behavior changes from the helper extraction.

Ran terminal command: $pyRoot='C:\Users\email\AppData\Local\Programs\Python\Python314'; $scriptsDir=Join-Path $pyRoot 'Scripts'; $env:PATH = "$pyRoot;$scriptsDir;" + $env:PATH; pytest tests/test_extract.py -q

Implemented the refactor to remove logger-selection duplication in extract.py.

### What changed
- Added shared helper:
  - `_resolve_logger(logger, chunk_number) -> logging.Logger | ChunkAdapter`
- Replaced duplicated logic in both:
  - `extract_grobid_chunk(...)`
  - `extract_pdfminer_chunk(...)`

Behavior is preserved:
- No `chunk_number` -> provided `logger` or default logger.
- `chunk_number` with no `logger` -> `get_chunk_logger(chunk_number)`.
- `chunk_number` with `logger` -> wraps provided logger in `ChunkAdapter`.

### Validation
- Ran: `pytest test_extract.py -q`
- Result: `18 passed, 1 skipped` (plus existing `RequestsDependencyWarning`)

File: docdown/stages/extract.py [hunk 2]
```diff
@@ -163,6 +168,47 @@ def extract_grobid_chunk(
...
+    if not chunk_path.exists() or not chunk_path.is_file():
+        raise PdfMinerError(f"Chunk PDF not found: {chunk_path}")
+
+    started_at = time.monotonic()
+
+    try:
+        text = pdfminer_extract_text(str(chunk_path))
+    except Exception as exc:
+        active_logger.error("pdfminer extraction failed for %s: %s", chunk_path.name, exc)
```
[the remark this answers is not in the document]
Comment (Frank-Yong) [reply docdown/stages/extract.py]: Ran terminal command: $pyRoot='C:\Users\email\AppData\Local\Programs\Python\Python314'; $scriptsDir=Join-Path $pyRoot 'Scripts'; $env:PATH = "$pyRoot;$scriptsDir;" + $env:PATH; pytest tests/test_extract.py -q

Updated as requested in extract.py:

- Changed pdfminer failure logging from:
  - `active_logger.error("pdfminer extraction failed for %s: %s", chunk_path.name, exc)`
- To:
  - `active_logger.exception("pdfminer extraction failed for %s", chunk_path.name)`

This now includes full traceback in logs, which is much more useful for debugging.

Validation:
- `pytest test_extract.py -q` -> `18 passed, 1 skipped` (plus existing `RequestsDependencyWarning`)

File: docdown/stages/extract.py [hunk 3]
```diff
@@ -163,6 +164,43 @@ def extract_grobid_chunk(
...
+    """Extract plain text from a chunk PDF using pdfminer and write UTF-8 text output."""
+
+    chunk_path = Path(chunk_pdf)
+    output_path = Path(output_text)
+
+    active_logger = _resolve_logger(logger, chunk_number)
+
+    if not chunk_path.exists() or not chunk_path.is_file():
+        raise PdfMinerError(f"Chunk PDF not found: {chunk_path}")
+
```
[the remark this answers is not in the document]
Comment (Frank-Yong) [reply docdown/stages/extract.py:183]: Ran terminal command: $pyRoot='C:\Users\email\AppData\Local\Programs\Python\Python314'; $scriptsDir=Join-Path $pyRoot 'Scripts'; $env:PATH = "$pyRoot;$scriptsDir;" + $env:PATH; pytest tests/test_extract.py -q

Added the missing unit test coverage for the pdfminer missing-file contract.

### Change made
In test_extract.py, added:

- `test_extract_pdfminer_chunk_rejects_missing_chunk_file`
  - Calls `extract_pdfminer_chunk` with a non-existent `chunk-0001.pdf`
  - Asserts `PdfMinerError`
  - Verifies message includes `Chunk PDF not found: ...chunk-0001.pdf`

### Validation
Ran:
- `pytest test_extract.py -q`

Result:
- `19 passed, 1 skipped` (plus existing `RequestsDependencyWarning`)

Made changes.
